File: PackageMaker/PackageMaker/Asset.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PackageMaker
{
    class Asset
    {
        private string name, path;
        private byte[] data;
        private UInt32 startByte;

        public Asset(string _path, byte[]_data)
        {
            path = _path;
            name = Utils.trimPathFromName(path);
            data = _data;
        }

        public void setStart(UInt32 _start)
        {
            startByte = _start;
        }

        public UInt32 getStart()
        {
            return startByte;
        }

        public byte[] getBytes()
        {
            return data;
        }

        public string getName()
        {
            return name;
        }

        public string getPath()
        {
            return path;
        }

        public UInt32 getSize()
        {
            return (UInt32)data.Length;
        }
    }
}

File: GE_Projekt_2/src/core/Core/ContentHandler/ContainerLoaders/Asset.cpp
#include "Asset.hpp"

Asset::Asset() : name("N/A"), startByte(0), size(0)
{
}

Asset::Asset(std::string _name, uint _startByte, uint _size) : name(_name), startByte(_startByte), size(_size)
{
}

Asset::~Asset()
{

}

std::string Asset::GetName()
{
	return name;
}

uint Asset::GetSize()
{
	return size;
}

uint Asset::GetStart()
{
	return startByte;
}